### PR TITLE
refactor(server): remove separate notifier process

### LINKED ISSUE #1796: Remove notifier.js
cc @jrgm @vladikoff

### PR DESCRIPTION
**Pull request by vladikoff**
cc @jrgm 

~~Work in progress...~~

Fixes #1796

### CODE REVIEW

File: lib/log.js [hunk 1]
```diff
@@ -150,7 +151,7 @@ Lug.prototype.notifyAttachedServices = function (name, request, data) {
           data: unbuffer(data)
         }
         e.data.metricsContext = metricsContextData
-        this.stdout.write(JSON.stringify(e) + '\n')
+        this.notifier.send(e)
```
Comment (seanmonstar) [inline lib/log.js]: Um, yes please!

**Comment by vladikoff** [comment]
@jrgm r? @mozilla/fxa-devs r?

File: lib/notifier.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+'use strict'
+
+/**
+ * This notifier is called by the logger via `notifyAttachedServices`
+ * to send notifications to Amazon SNS/SQS.
+ */
+const AWS = require('aws-sdk')
+const config = require('../config')
+
+const snsTopicArn = config.get('snsTopicArn')
```
Comment (vbudhram) [inline lib/notifier.js]: The config value name `snsTopicArn` doesn't mention what the topic is used for. Maybe `notifierSnsTopicArn`?

File: test/local/notifier.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+  beforeEach(() => {
+    log.error.reset()
+    log.trace.reset()
+  })
+
+  it('works with sns configuration', () => {
+    const config = {
+      get: (key) => {
+        if(key === 'snsTopicArn') {
```
Comment (vbudhram) [inline test/local/notifier.js]: nit space after `if`

File: test/local/notifier.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      },
+      success: true
+    })
+    assert.equal(log.error.called, false)
+  })
+
+  it('works with disabled configuration', () => {
+    const config = {
+      get: (key) => {
+        if(key === 'snsTopicArn') {
```
Comment (vbudhram) [inline test/local/notifier.js]: nit space after `if`

File: lib/notifier.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  return {
+    send: (event, callback) => {
+      const msg = event.data || {}
+      msg.event = event.event
+      sns.publish({
+        TopicArn: snsTopicArn,
+        Message: JSON.stringify(msg)
+      }, (err, data) => {
+        if (err) {
+          log.error({op: 'Notifier.sent', err: err})
```
Comment (vbudhram) [inline lib/notifier.js]: Previous file used `Notifier.published`, do we have any graphs or metrics that would need updating to use `Notifier.sent`?

File: lib/notifier.js (hunk 2 above)
Comment (vladikoff) [inline lib/notifier.js]: That will probably require changes to existing configs in puppet, etc. So I'd rather just leave it alone, I can rename the local variable though

File: lib/notifier.js (hunk 5 above)
Comment (vladikoff) [inline lib/notifier.js]: Good idea. reverting to the older name...

**Comment by vladikoff** [comment]
Updated

**Pull request merged**